src/Rules/Properties/AccessPropertiesRule.php

```diff
@@ -41,101 +43,104 @@ public function getNodeType(): string
 	{
-		if (!$node->name instanceof \PhpParser\Node\Identifier) {
-			return [];
+		if ($node->name instanceof \PhpParser\Node\Identifier) {
+			$names = [$node->name->name];
+		} else {
+			$names = array_map(static function (ConstantStringType $string): string {
+				return $string->getValue();
+			}, TypeUtils::getConstantStrings($scope->getType($node->name)));
 		}
 
-		$name = $node->name->name;
-		$typeResult = $this->ruleLevelHelper->findTypeToCheck(
-			$scope,
-			$node->var,
-			sprintf('Access to property $%s on an unknown class %%s.', $name),
-			static function (Type $type) use ($name): bool {
-				return $type->canAccessProperties()->yes() && $type->hasProperty($name)->yes();
+		$errors = [];
+		foreach ($names as $name) {
+			$typeResult = $this->ruleLevelHelper->findTypeToCheck(
+				$scope,
+				$node->var,
+				sprintf('Access to property $%s on an unknown class %%s.', $name),
+				static function (Type $type) use ($name): bool {
+					return $type->canAccessProperties()->yes() && $type->hasProperty($name)->yes();
+				}
+			);
+			$type = $typeResult->getType();
+			if ($type instanceof ErrorType) {
+				$errors = array_merge($errors, $typeResult->getUnknownClassErrors());
+				continue;
 			}
-		);
-		$type = $typeResult->getType();
-		if ($type instanceof ErrorType) {
-			return $typeResult->getUnknownClassErrors();
-		}
 
-		if ($scope->isInExpressionAssign($node)) {
-			return [];
-		}
+			if ($scope->isInExpressionAssign($node)) {
+				continue;
+			}
 
-		if (!$type->canAccessProperties()->yes()) {
-			return [
-				RuleErrorBuilder::message(sprintf(
+			if (!$type->canAccessProperties()->yes()) {
+				$errors[] = RuleErrorBuilder::message(sprintf(
 					'Cannot access property $%s on %s.',
 					$name,
 					$type->describe(VerbosityLevel::typeOnly())
-				))->build(),
-			];
-		}
-
-		if (!$type->hasProperty($name)->yes()) {
-			if ($scope->isSpecified($node)) {
-				return [];
+				))->build();
+				continue;
 			}
 
-			$classNames = $typeResult->getReferencedClasses();
-			if (!$this->reportMagicProperties) {
-				foreach ($classNames as $className) {
-					if (!$this->reflectionProvider->hasClass($className)) {
-						continue;
-					}
+			if (!$type->hasProperty($name)->yes()) {
+				if ($scope->isSpecified($node)) {
+					continue;
+				}
 
-					$classReflection = $this->reflectionProvider->getClass($className);
-					if (
-						$classReflection->hasNativeMethod('__get')
-						|| $classReflection->hasNativeMethod('__set')
-					) {
-						return [];
+				$classNames = $typeResult->getReferencedClasses();
+				if (!$this->reportMagicProperties) {
+					foreach ($classNames as $className) {
+						if (!$this->reflectionProvider->hasClass($className)) {
+							continue;
+						}
+
+						$classReflection = $this->reflectionProvider->getClass($className);
+						if (
+							$classReflection->hasNativeMethod('__get')
+							|| $classReflection->hasNativeMethod('__set')
+						) {
+							continue 2;
```

Comment:
Why continue 2 in this case?

Reply:
It was already "continue" before we added a foreach loop around the code, so I added it to keep the same logic.